### PR TITLE
Allow newsletter signup module to have two optional name fields & update newsletter module on Bannered Campaign Page

### PR DESCRIPTION
**Pull request by mmmavis**
Hey @danielfmiranda , apologies for another PR review request. This PR has more changes involved but I promise the future ones will be quick and easy to review. 😁 Thanks again for your time and please let me know if I can help clarify anything.

Note: Reviewing this PR commit by commit should make the code review process easier.

### Updated the existing components so they can accommodate more functionalities for optional fields
  - If "Ask Name" field on `signup` snippet is set to `true`, signup form should 
    - render two additional fields (first name and last name) by default
    - country dropdown should show by default
    - language dropdown should show by default
    - If `showCountryFieldByDefault` is true, show country field on the initial render, otherwise the this field will be toggled when email field is on focus/input
    - If `showLanguageFieldByDefault` is true, show langue field on the initial render, otherwise the this field will be toggled when email field is on focus/input 
  - If "Ask Name" field on `signup` snippet is set to `false`, signup form should 
    - not render first name field
    - not render last name field
    - If `showCountryFieldByDefault` is true, show country field on the initial render, otherwise the this field will be toggled when email field is on focus/input
    - If `showLanguageFieldByDefault` is true, show langue field on the initial render, otherwise the this field will be toggled when email field is on focus/input
  

Related ticket: #11668

# To QA

### With "ask name" set to "true"
Staging: https://foundation.mofostaging.net/en/campaigns/test-signup-ask-name-set-to-true/
Review app: https://foundation-s-11668-news-wfq8cj.herokuapp.com/en/campaigns/initial-test-bannered-campaign-with-fixed-title/

### With "ask name" set to "false"
Staging: https://foundation.mofostaging.net/en/campaigns/test-signup-ask-name-set-to-false/
Review app: Same link but please [edit this page's Signup snippet](https://foundation-s-11668-news-wfq8cj.herokuapp.com/cms/snippets/wagtailpages/signup/edit/8/)'s `ask name` checkbox to `false`. See screenshot below. (CMS creds: admin2/admin2password)

image

### CODE REVIEW

File: source/js/components/newsletter-signup/organisms/default-layout-signup.jsx [hunk 1]
```diff
@@ -212,33 +215,37 @@ class DefaultSignupForm extends Component {
     );
   }
 
-  renderAdditionalFields() {
```
Comment (mmmavis) [inline source/js/components/newsletter-signup/organisms/default-layout-signup.jsx:155]: break into two methods: `renderCountryField` and `renderLanguageField`

File: network-api/networkapi/wagtailpages/templates/wagtailpages/tags/cta.html [hunk 2]
```diff
@@ -1,5 +1,3 @@
-{{ cta.type }}
```
Comment (mmmavis) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/tags/cta.html:1]: This doesn't do anything. I'm guessing someone had this for testing purpose but forgot to remove it before merging the PR.

File: source/js/components/newsletter-signup/organisms/default-layout-signup.jsx [hunk 3]
```diff
@@ -90,21 +86,32 @@ class DefaultSignupForm extends Component {
...
   }
 
   handleCountryChange(event) {
-    this.updateFormFieldValue("country", event.target.value);
+    this.updateFormFieldValue(event.target.name, event.target.value);
   }
 
   handleLanguageChange(event) {
-    this.updateFormFieldValue("language", event.target.value);
+    this.updateFormFieldValue(event.target.name, event.target.value);
```
Comment (danielfmiranda) [inline source/js/components/newsletter-signup/organisms/default-layout-signup.jsx:122]: Just giving Kudos to what I think is a great idea!

**Review by danielfmiranda — APPROVED** [review]
Hi @mmmavis!

Thanks for taking this work on 🙌 and for calling out that reviewing commit-by-commit may be easier than reviewing all of the code changes at once. I took your suggested approach while reviewing this PR and can agree that it did make it easier to understand the changes.

After taking a look at the code and also testing on the review app, I can confirm that everything is working as expected.

Approved! 👍

**Pull request merged**